### PR TITLE
Thuang 518 download part 2

### PR DESCRIPTION
**Pull request by tihuan**
1. Rebased from Trent's Part 1 PR!

1. Add two smoke tests to assert clicking on the "download" button in the column triggers the modal, and also assert the user flow of selecting a data format and the download link exists 

1. Update `backend/config/corpora-api.yml` to match actual response keys

1. Remove temporary fixture: `frontend/src/common/fixtures/projects.ts`, since now we have local database and dev API endpoint to use

1. Add functionality for download modal - will comment in code for clarity! 

gif

QA Steps:
1. Make sure your `frontend/src/configs/configs.js` points to `API_URL: "https://api.dev.corpora.cziscience.com",`, so you have more datasets to test
2. Start FE server! `npm run develop`
3. Click on any of the "Download" button in the Dataset rows, and it should trigger a modal
4. Select a data format
5. Either copy CURL and run the command in the terminal OR click "Download" to start downloading
6. That's it :D

Thank you!!

### CODE REVIEW

**Comment by tihuan** [comment]
I think I'll wait till @Bento007 's PR is merged into the feature branch then resolve all the conflicts at once!

**Comment by tihuan** [comment]
I'll keep this PR as a draft until Trent's PR is merged into the feature branch!

File: frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/style.ts
Comment (tihuan) [inline frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/style.ts:49]: party trick to use `filter: brightness()` to "highlight" the element on hover!

https://css-tricks.com/using-a-brightness-filter-to-generically-highlight-content/

File: frontend/src/components/ProjectList/components/Dataset/common/style.ts
Comment (tihuan) [inline frontend/src/components/ProjectList/components/Dataset/common/style.ts:17]: consolidate button style for both "View" and "Download" here

File: frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/style.ts
Comment (tihuan) [inline frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/style.ts:5]: consolidated!

File: frontend/src/components/ProjectList/components/Dataset/components/ViewDataset/style.ts
Comment (tihuan) [inline frontend/src/components/ProjectList/components/Dataset/components/ViewDataset/style.ts:5]: same here!

File: frontend/src/components/common/theme.ts
Comment (tihuan) [inline frontend/src/components/common/theme.ts:11]: color constants

File: frontend/tests/features/homepage.test.ts
Comment (tihuan) [inline frontend/tests/features/homepage.test.ts:25]: added smoke tests

File: frontend/jest-playwright.config.js
Comment (Bento007) [reply frontend/jest-playwright.config.js:16]: We should upload a small fake file to test

**Review by Bento007 — APPROVED** [review]
Very nice!

File: frontend/src/components/ProjectList/components/Dataset/common/style.ts
Comment (pdugan20) [reply frontend/src/components/ProjectList/components/Dataset/common/style.ts:17]: This is great.

File: frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/style.ts
Comment (pdugan20) [inline frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/style.ts:10]: Might want to standardize this like BLUE at some point?

**Review by pdugan20 — APPROVED** [review]
View code all looks great. Thanks for knocking this out Timmy!

File: frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/style.ts
Comment (tihuan) [reply frontend/src/components/ProjectList/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/style.ts]: Ooh thanks for catching this, Pat! Yes we have enum for `GREY` already, so I just added this one as `VERY_LIGHT`, according to https://www.htmlcsscolor.com/hex/CDCDCD#:~:text=%23CDCDCD%20(or%200xCDCDCD)%20is,of%20max%20value%20%3D%20765 😄 

Thank you!

File: frontend/jest-playwright.config.js
Comment (tihuan) [reply frontend/jest-playwright.config.js:16]: ahhh that's a great idea! Yeah I'll spend some time today to play with the download API a little more, given that in theory it supports reading the downloading file's stream, so we shouldn't need to wait for the whole file to complete lol I'll check with Playwright team on this too!

**Comment by tihuan** [comment]
Thanks so much for the great reviews both!! 🤩 🙏 I added auto merge for this to go into feature branch, and then I'll merge feature branch into `main` and move the tickets 👌 

Happy Thursday!

**Comment by tihuan** [comment]
Hey @Bento007 ! Need help with the BE for smoke test lol

Looks like something is going on here, but running local BE server tests seems to be fine on my laptop. Do you think it has something to do with the AWS role GH action assumes?

THANK YOU!

>1210: ERROR:corpora.common.entities.dataset_asset:Failed to retrieve meta data for 'ParseResult(scheme='s3', netloc='bogus_bucket', path='/test_s3_uri.h5ad', params='', query='', fragment='')'.
Traceback (most recent call last):
  File "/home/runner/work/corpora-data-portal/corpora-data-portal/backend/chalice/api_server/chalicelib/corpora/common/entities/dataset_asset.py", line 48, in get_file_size
    response = self.s3.head_object(Bucket=self.bucket_name, Key=self.key_name)
corpora-api-test - INFO - Response: {'headers': {'X-AWS-REQUEST-ID': '63081609-b4c9-4124-a4ec-ca0a631b06a6', 'Content-Type': 'application/problem+json', 'Content-Length': '159'}, 'multiValueHeaders': {}, 'statusCode': 500}
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/botocore/client.py", line 337, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/botocore/client.py", line 656, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found

Screen Shot 2020-09-17 at 10 40 54 AM

**Pull request merged**
